### ISSUE TITLE
NetCore11 Deserialize Not Working

### ISSUE THREAD

**Issue by SPEMoorthy** [esskar/Serialize.Linq#100]
Hi.
I am Trying to execute the code in **NetCore11** version

            Expression<Func<Mfr, bool>>  Predicate = p => p.MfrId.ToString().Contains("ABC") || p.MfrName.Contains("ABC");
            ExpressionSerializer es = new ExpressionSerializer(new BinarySerializer());
            byte[] byteArr = es.SerializeBinary(Predicate);
            //Convert Back
            Expression<Func<Mfr, bool>> Predicate1 = (Expression<Func<Mfr, bool>>)es.DeserializeBinary(byteArr);

**We are getting error. "Failed to serialize 'SITSLN.GSRetailCore.Models.Mfr' to a type object." But the same code is working in Net 45. Also Tried other types (JSON and XML) :-(**

Stack Trace For your Ref
```
            at Serialize.Linq.Nodes.TypeNode.ToType(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\TypeNode.cs:line 96
           at Serialize.Linq.Nodes.MemberNode`1.GetDeclaringType(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\MemberNode.cs:line 104
           at Serialize.Linq.Nodes.MemberNode`1.ToMemberInfo(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\MemberNode.cs:line 129
           at Serialize.Linq.Nodes.MemberExpressionNode.ToExpression(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\MemberExpressionNode.cs:line 59
           at Serialize.Linq.Nodes.MethodCallExpressionNode.ToExpression(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\MethodCallExpressionNode.cs:line 72
           at Serialize.Linq.Nodes.MethodCallExpressionNode.ToExpression(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\MethodCallExpressionNode.cs:line 72
           at Serialize.Linq.Nodes.BinaryExpressionNode.ToExpression(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\BinaryExpressionNode.cs:line 145
           at Serialize.Linq.Nodes.LambdaExpressionNode.ToExpression(IExpressionContext context) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\LambdaExpressionNode.cs:line 67
           at Serialize.Linq.Nodes.ExpressionNode.ToExpression() in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Nodes\ExpressionNode.cs:line 141
           at Serialize.Linq.Serializers.ExpressionSerializer.DeserializeBinary(Byte[] bytes) in D:\Source\Repos\Serialize.Linq\src\Serialize.Linq\Serializers\ExpressionSerializer.cs:line 108
           at GSRetailWeb.Api.Masters.MfrController.<SearchMfr>d__2.MoveNext() in D:\Source\Repos\GSRetail\GSRetailWeb\Api\Masters\MfrController.cs:line 46
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
           at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionMethodAsync>d__12.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeNextActionFilterAsync>d__10.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
           at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
           at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeNextResourceFilter>d__22.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
           at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
           at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.<Invoke>d__6.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at SITSLN.GSRetailWeb.Middlewares.ErrorWrappingMiddleware.<Invoke>d__3.MoveNext() in D:\Source\Repos\GSRetail\GSRetailWeb\Middlewares\ErrorWrappingMiddleware.cs:line 28
```

**Comment by yatli** [comment]
We're experiencing similar problems, and @chaosddp discovered that if your assembly name differs from your package id, assembly loading will fail, here: https://github.com/esskar/Serialize.Linq/blob/b2da08d4d005aef0a02915b274223baa2aebb2e9/src/Serialize.Linq.NetCore11/Internals/AppDomain.cs#L32

Linking our issue:
https://github.com/Microsoft/GraphEngine/issues/168

**Comment by chaosddp** [comment]
Or if your object's assembly does not depend on any library that the name starts with "System", then you will get same exception: https://github.com/esskar/Serialize.Linq/blob/b2da08d4d005aef0a02915b274223baa2aebb2e9/src/Serialize.Linq.NetCore11/Internals/AppDomain.cs#L52

**Comment by esskar** [comment]
I just published version 1.7.0 that supports .net standard 1.3 and 2.0
does this help for you?

**Comment by SPEMoorthy** [comment]
Thanks. Let me check and update you.
